File: src/org/usfirst/frc5293/commands/autonomous/TimedCommand.java
package org.usfirst.frc5293.commands.autonomous;

import edu.wpi.first.wpilibj.Timer;
import edu.wpi.first.wpilibj.command.Command;

public abstract class TimedCommand extends Command {

    private final double seconds;
    private Timer timer;

    protected TimedCommand(double seconds) {
        super();
        this.seconds = seconds;
    }

    @Override
    protected void initialize() {
        timer = new Timer();
        timer.start();
    }

    @Override
    protected boolean isFinished() {
        return timer.hasPeriodPassed(seconds);
    }

    @Override
    protected void end() {
        timer.stop();
    }

    @Override
    protected void interrupted() {
        timer.start();
    }
}

File: src/org/usfirst/frc5293/commands/util/TimedCommand.kt
package org.usfirst.frc5293.commands.util

import edu.wpi.first.wpilibj.Timer
import edu.wpi.first.wpilibj.command.Command

abstract class TimedCommand protected constructor(private val seconds: Double) : Command() {
    private var timer: Timer? = null

    override fun initialize() {
        timer = Timer()
        timer!!.start()
    }

    override fun isFinished(): Boolean {
        return timer!!.hasPeriodPassed(seconds)
    }

    override fun end() {
        timer!!.stop()
    }

    override fun interrupted() {
        timer!!.start()
    }
}
